keras_nlp/models/bart/bart_seq_2_seq_lm.py

```diff
@@ -0,0 +1,717 @@
+                x = tf.convert_to_tensor(x)
+
+            if isinstance(x, tf.Tensor) and x.shape.rank == 0:
+                x_is_scalar = True
+                x = x[tf.newaxis]
+
+            return x, x_is_scalar
+
+        if isinstance(inputs, dict):
+            for key in inputs:
+                inputs[key], input_is_scalar = normalize(inputs[key])
+        else:
+            inputs, input_is_scalar = normalize(inputs)
+
+        # We avoid converting to a dataset purely for speed, for a single batch
+        # of input, creating a dataset would add significant overhead.
+        return [inputs], input_is_scalar
+
+    def _normalize_generate_outputs(
+        self,
+        outputs,
+        input_is_scalar,
+    ):
+        """Normalizes user output from the generate function.
+
+        This function converts the output to numpy (for integer output), or
+        python strings (for string output). If a batch dimension was added to
+        the input, it is removed from the output (so generate can be string in,
+        string out).
+        """
+
+        def normalize(x):
+            x = tf.concat(x, axis=0)
+            x = tf.squeeze(x, 0) if input_is_scalar else x
+            is_string = x.dtype == tf.string
+            # Convert outputs to a friendly pythonic type. For numerical outputs
+            # that is numpy, for string outputs that is `list` and `str`.
+            return tensor_to_string_list(x) if is_string else x.numpy()
+
+        if isinstance(outputs[0], dict):
+            return {
+                "decoder_token_ids": normalize(
+                    [x["decoder_token_ids"] for x in outputs]
+                ),
+                "decoder_padding_mask": normalize(
+                    [x["decoder_padding_mask"] for x in outputs]
+                ),
+            }
+        return normalize([x for x in outputs])
+
+    def generate(
+        self,
+        inputs,
+        max_length=None,
+    ):
+        """Generates text conditioned on the encoder inputs.
+
+        This method generates text based on given `inputs`. The sampling method
+        used for generation can be set in the `compile` method.
+
+        If `inputs` is a `tf.data.Dataset`, outputs will be generated
+        "batch-by-batch" and concatenated. Otherwise, all inputs will be handled
+        as a single batch.
+
+        If a `preprocessor` is attached to the model, `inputs` is a dictionary
+        containing `encoder_text` and `decoder_text` keys, with strings for
+        values. The returned sequences will be strings. Otherwise, `inputs`
+        should be preprocessed before calling `generate()` - `inputs` will
+        contain key `encoder_token_ids`, `encoder_padding_mask`,
+        `decoder_token_ids` and `decoder_padding_mask`, and the returned
+        sequences will also be token IDs.
+
+        Args:
+            inputs: a Python string, a list of Python strings, a dense tensor of
```

Comment:
This feels like way too much complexity at once, so let's find a way to say it more simply. Something like...

A single input, batch of inputs, or `tf.data.Dataset` of batched inputs. If a `preprocessor` is attached,
each input can be a simple string, for basic conditional generation, or a dictionary with keys
`"encoder_text"` and `"decoder_text"` to specify a prompt. If a `preprocessor` is not attached, 
input batches should have the same structure as when directly calling the model.

Alternately we could try something like [model.fit](https://github.com/keras-team/keras/blob/v2.12.0/keras/engine/training.py#L1363-L1382), but even then I might do this as a follow up to be able to review it more clearly on its own.